src/messages/mod.rs

```diff
@@ -228,12 +228,22 @@ impl FullSecurityMetadata {
             TrustStatus::ProofInvalid => return Err(RoutingError::UntrustedMessage),
         };
 
+        self.verify_signature(content, public_key)?;
+        Ok(VerifyStatus::Full)
+    }
+
+    #[cfg_attr(feature = "mock_base", allow(clippy::trivially_copy_pass_by_ref))]
+    fn verify_signature(
```

Comment:
I guess we can revert that function since it is not use and does not seem to bring much.